src/r2mm/downloading/BetterThunderstoreDownloader.ts

```diff
@@ -86,35 +87,23 @@ export default class BetterThunderstoreDownloader extends ThunderstoreDownloader
                            ignoreCache: boolean,
                            callback: (progress: number, modName: string, status: number, err: R2Error | null) => void,
                            completedCallback: (modList: ThunderstoreCombo[]) => void) {
+
+        const modList = await ProfileModList.getModList(profile);
+        if (modList instanceof R2Error) {
+            return callback(0, mod.getName(), StatusEnum.FAILURE, modList);
+        }
+
         let dependencies: ThunderstoreCombo[] = [];
         await this.buildDependencySet(modVersion, dependencies, DependencySetBuilderMode.USE_EXACT_VERSION);
         this.sortDependencyOrder(dependencies);
         const combo = new ThunderstoreCombo();
         combo.setMod(mod);
         combo.setVersion(modVersion);
-        let downloadCount = 0;
-
-        const modList = await ProfileModList.getModList(profile);
-        if (modList instanceof R2Error) {
-            return callback(0, mod.getName(), StatusEnum.FAILURE, modList);
-        }
 
+        dependencies = await this.determineDependencyVersions(dependencies, combo, modVersion, modList);
         let downloadableDependencySize = this.calculateInitialDownloadSize(dependencies);
 
-        // Determine if modpack
-        // If modpack, use specified dependencies as retrieved above
-        // If not, get latest version of dependencies.
-        let isModpack = combo.getMod().getCategories().find(value => value === "Modpacks") !== undefined;
-        if (!isModpack) {
-            // If not modpack, get latest
-            dependencies = [];
-            await this.buildDependencySet(modVersion, dependencies, DependencySetBuilderMode.USE_LATEST_VERSION);
-            this.sortDependencyOrder(dependencies);
-            // #270: Remove already-installed dependencies to prevent updating.
-            dependencies = dependencies.filter(dep => modList.find(installed => installed.getName() === dep.getMod().getFullName()) === undefined);
-            downloadableDependencySize = this.calculateInitialDownloadSize(dependencies);
```

Comment:
~Seems you left this line out from your new method, which I think would normally be a bug. But, `calculateInitialDownloadSize` is stupid and its docstring is a lie. IMO we should yeet `downloadableDependencySize` and just refer to `dependencies.length`~ ah well it was addressed in the next commit